src/components/Home/Aditya/Week4/pages/Week1/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import CreateTaskCard from './components/CreateTaskCard/CreateTaskCard';
+import { useState, useEffect } from 'react';
+import { TaskDetails } from './interfaces/TaskDetailsInterface';
+import TaskList from './components/TaskList/TaskList';
+import { AppWrapper } from './styles';
+
+const App: React.FC = () => {
+  const getLocalTasks = (): TaskDetails[] => {
+    return JSON.parse(localStorage.getItem('tasks') ?? '[]');
+  };
+
+  const setLocalTasks = (tasks: TaskDetails[]): void => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  };
+
+  const [tasks, setTasks] = useState<TaskDetails[]>(getLocalTasks());
+  const [currentFilter, setCurrentFilter] = useState<string>('All Tasks');
+  const [showTasks, setShowTasks] = useState<TaskDetails[]>(getLocalTasks());
+
+  useEffect(() => {
+    if (currentFilter === 'Due Closest' || currentFilter === 'Due Farthest') {
```

Comment:
use it inside switch you can hve as
case "Due Closest":
case "Due Farthest":
const tempTasks = [...tasks];
      const sortedTasks = tempTasks.sort((task1, task2) => {
        const date1 = new Date(task1.date);
        const date2 = new Date(task2.date);
        return date1.getTime() - date2.getTime();
      });
      currentFilter === 'Due Farthest' && sortedTasks.reverse();
      setShowTasks(sortedTasks);